From feec11ae7278e90f4e72d22c0f35eb86d16dc2dd Mon Sep 17 00:00:00 2001
From: Enrico Jorns <ejo@pengutronix.de>
Date: Tue, 9 Jan 2018 18:09:53 +0100
Subject: [PATCH 28/44] src/update_handler.c: add support for catar images

When using casync-based bundles, catar images will reach a much higher
matching rate for the chunking algorithm than classical (uncompressed)
tar archives.

Signed-off-by: Enrico Jorns <ejo@pengutronix.de>
---
 src/update_handler.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/update_handler.c b/src/update_handler.c
index ef5db1b..688123a 100644
--- a/src/update_handler.c
+++ b/src/update_handler.c
@@ -531,6 +531,8 @@ static gboolean unpack_archive(RaucImage *image, gchar *dest, GError **error)
 {
 	if (g_str_has_suffix(image->filename, ".caidx" ))
 		return casync_extract_image(image, dest, error);
+	else if (g_str_has_suffix(image->filename, ".catar" ))
+		return casync_extract_image(image, dest, error);
 	else
 		return untar_image(image, dest, error);
 }
@@ -1104,6 +1106,7 @@ RaucUpdatePair updatepairs[] = {
 	{"*.ext4", "raw", img_to_raw_handler},
 	{"*.vfat", "raw", img_to_raw_handler},
 	{"*.tar*", "ext4", archive_to_ext4_handler},
+	{"*.catar", "ext4", archive_to_ext4_handler},
 	{"*.tar*", "ubifs", archive_to_ubifs_handler},
 	{"*.tar*", "vfat", archive_to_vfat_handler},
 	{"*.ubifs", "ubivol", img_to_ubivol_handler},
-- 
2.14.1

